compiler/il/OMRBlock.cpp

```diff
@@ -973,6 +973,12 @@ static bool checkStoreRegNodeListForNode(TR::Node *passThroughNode, List<TR::Nod
 
 /*
  * Analyze the register dependency and update the list of unavailable registers.
+ * It examines the node under passthrough children of the glRegDeps and does the following.
+ *    If node requires uncommoning and we do not have replacement yet, looks into the list
```

Comment:
When might a node require uncommoning ? 

Pass through nodes under GlRegDeps typically appear towards the end of the basic block, e.g. on some exit out of a block such as a branch or BBEnd. Since there can be more than one exit out of an extended basic block, which GlRegDeps is being examined here ?

Reply:
Basically after analyzing nodes before the split point, it checks the Nodes under passthrough nodes after split point. While walking tree-tops after split point it tries to utilize very first PassThrough it has encountered that references node and tries to utilize the used register if it is available. 

Trying to summarize possible cases with PassThrough nodes here. 

Case - 1 : 
```
n1n BBStart
...
// Split Point
...
n2n aRegStore Rx
n3n   =>nodeX
...
n4n BBEnd
n5n   GlRegDeps ()
n6n      PassThrough Rx
n3n      => nodeX
```
This is the simplest case, if `Rx` is available, it will use it to store `nodeX` before split point and changes all the reference of `nodeX` with the `aRegLoad Rx`. Nothing requires to be done if `Rx` is not available. Splitter would use other available register or temp slot for `nodeX` and all the references (including under `aRegStore`) of `nodeX` after split point will be changed by load from the chosen register or created temp slot. 

Case - 2 : 
```
n1n BBStart
...
n2n aRegStore Rx
n3n    => node X
// Split Point
...
n4n BBEnd
n5n   GlRegDeps ()
n6n      PassThrough Rx
n3n      => nodeX
```
Similar to Case - 1, it will use `Rx` for `nodeX` if it is available. As in this case, `aRegStore` is before split point, it does not need to add a store node.  But if Rx is not available then splitter might use another register or temp slot for `nodeX`, in this case it needs to add `aRegStore Rx` before the PassThrough as following. 
```
// SplitPoint
...
n7n aRegStore Rx
n3n    => nodeX
n4n BBEnd
n5n   GlRegDeps ()
n6n      PassThrough Rx
n3n      => nodeX
```

Analyzing PassThrough after split point is done mainly to minimize spillings (If we end up using different register then the one used under PassThrough, we would have spillings to free up the register). 
Also if the register used by PassThrough is used for other node, then we need to make sure that node referenced under PassThrough node is stored into same register as used by PassThrough node in newly created block before the exit edge which has that PassThrough node.